From b38d25c5553e65f684e77dd7fc08464754da9dac Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Fri, 23 Feb 2024 18:30:55 +0100
Subject: [PATCH 0666/1695] dts: stm32mp15: add operating point nodes and
 configuration

Add the OPP table configuration for the STM32MP15 platform
and its associated configuration depending to SoC version.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 134212714c82c1d7f69baf5e270b3811a0cd5065 dts: stm32mp15: add operating point nodes and configuration]
Change-Id: I7fb56d83514bca921f522cc09d9fd0033608fab7
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362692
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp151.dtsi  | 31 ++++++++++++++++++++++++++++++
 core/arch/arm/dts/stm32mp15xd.dtsi |  1 -
 2 files changed, 31 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index 6f39c0cbe..cb0f16c0d 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -20,6 +20,11 @@
 			clock-frequency = <650000000>;
 			device_type = "cpu";
 			reg = <0>;
+			clocks = <&rcc CK_MPU>;
+			clock-names = "cpu";
+			operating-points-v2 = <&cpu0_opp_table>;
+			nvmem-cells = <&part_number_otp>;
+			nvmem-cell-names = "part_number";
 		};
 	};
 
@@ -30,6 +35,32 @@
 		interrupt-parent = <&intc>;
 	};
 
+	cpu0_opp_table: cpu0-opp-table {
+		compatible = "operating-points-v2";
+		opp-shared;
+
+		opp-400000000 {
+			opp-hz = /bits/ 64 <400000000>;
+			opp-microvolt = <1200000>;
+			opp-supported-hw = <0x2>;
+			opp-suspend;
+		};
+
+		opp-650000000 {
+			opp-hz = /bits/ 64 <650000000>;
+			opp-microvolt = <1200000>;
+			opp-supported-hw = <0x1>;
+			st,opp-default;
+		};
+
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <1350000>;
+			opp-supported-hw = <0x2>;
+			st,opp-default;
+		};
+	};
+
 	psci {
 		compatible = "arm,psci-1.0";
 		method = "smc";
diff --git a/core/arch/arm/dts/stm32mp15xd.dtsi b/core/arch/arm/dts/stm32mp15xd.dtsi
index c0d507a27..20e52cd27 100644
--- a/core/arch/arm/dts/stm32mp15xd.dtsi
+++ b/core/arch/arm/dts/stm32mp15xd.dtsi
@@ -3,4 +3,3 @@
  * Copyright (C) STMicroelectronics 2019 - All Rights Reserved
  * Author: Alexandre Torgue <alexandre.torgue@st.com> for STMicroelectronics.
  */
-
-- 
2.47.2

